src/main/java/bookface/logic/parser/delete/DeleteBookArgumentsParser.java

```diff
@@ -0,0 +1,28 @@
+package bookface.logic.parser.delete;
+
+import bookface.commons.core.Messages;
+import bookface.commons.core.index.Index;
+import bookface.logic.commands.delete.DeleteBookCommand;
+import bookface.logic.parser.ArgumentsParsable;
+import bookface.logic.parser.ParserUtil;
+import bookface.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates the relevant new AddCommand object for the relevant entity to be added
+ */
+public class DeleteBookArgumentsParser implements ArgumentsParsable<DeleteBookCommand> {
+    /**
+     * Parses the given arguments in the context of the DeleteCommand
+     * and returns a DeleteCommand object for execution.
+     */
+    @Override
+    public DeleteBookCommand parse(String args) throws ParseException {
```

Comment:
This is just a generic comment, this sort of check if the index is parseable is a common functionality across many commands. I think we can explore ways to abstract this out later